src/Microsoft.DotNet.Build.Tasks/MakeResourcesPriFile.cs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Xml.Linq;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class MakeResourcesPriFile : Task
+    {
+        private string _reswListFile;
+        private string _priListFile;
+        private string _tempFolder;
+        private string _modifiedConfigFile;
+
+        [Required]
+        public ITaskItem[] ResWFiles { get; set; }
+
+        [Required]
+        public string RunnersPriFilePath { get; set; }
+
+        [Required]
+        public string NewPriFilePath { get; set; }
+
+        [Required]
+        public string MakePriExecutablePath { get; set; }
+
+        [Required]
+        public string ConfigFilePath { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                Initialize();                
+                WritePriListFile();
+                WriteReswListFile();
+                UpdateConfigFile();
+                MergePriFiles();
+                Cleanup();
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e, showStackTrace: true);
+                return false; // fail the task
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private void Initialize()
+        {
+            if (!File.Exists(ConfigFilePath))
+            {
+                throw new ArgumentException($"MakePri config file: {ConfigFilePath} was not found.");
+            }
+
+            if (!File.Exists(RunnersPriFilePath))
+            {
+                throw new ArgumentException($"App runner resources.pri file: {RunnersPriFilePath} was not found.");
+            }
+
+            if (!File.Exists(MakePriExecutablePath))
+            {
+                throw new ArgumentException($"MakePri.exe was not found in path: {MakePriExecutablePath}");
+            }
+
+            _tempFolder = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName(), "makeprifiles");
```

Comment:
No reason in particular. I just wanted to use some directory that wasn't in the common path where I would be adding files that are used just once by makepri to the repo. Is there a way to get the intermediate directory inside the task or should I pass it as a parameter?